src/engine/loopingcontrol.cpp

```diff
@@ -1153,22 +1114,20 @@ void LoopingControl::slotBeatJumpBackward(double pressed) {
 }
 
 void LoopingControl::slotLoopMove(double beats) {
-    if (!m_pTrack || !m_pBeats) {
+    if (m_pTrack == nullptr || m_pBeats == nullptr || beats == 0) {
         return;
     }
     LoopSamples loopSamples = m_loopSamples.getValue();
     if (loopSamples.start == kNoTrigger || loopSamples.end == kNoTrigger) {
         return;
     }
 
-    double dPosition = getCurrentSample();
-    double dBeatLength;
-    if (BpmControl::getBeatContext(m_pBeats, dPosition,
-                                   NULL, NULL, &dBeatLength, NULL)) {
+    if (BpmControl::getBeatContext(m_pBeats, getCurrentSample(),
+                                   nullptr, nullptr, nullptr, nullptr)) {
         int old_loop_in = loopSamples.start;
         int old_loop_out = loopSamples.end;
-        int new_loop_in = old_loop_in + (beats * dBeatLength);
-        int new_loop_out = old_loop_out + (beats * dBeatLength);
+        int new_loop_in = m_pBeats->findNBeatsFromSample(old_loop_in, beats);
+        int new_loop_out = m_pBeats->findNBeatsFromSample(old_loop_out, beats);
```

Comment:
I adjusted the BPM used in the test to a more realistic number (120). I think the rounding issue may have only appeared because the test used such a high BPM that rounding beats to the nearest frame could produce different results... I may be misunderstanding that though...